### DISCUSSION TITLE
Unquote

### DISCUSSION THREAD

**Discussion by djcaesar9114** [atom/atom#23166] [Ideas]
Hello,
First of all thank you for this beautiful tool, I can't code without Atom!
When I select a text and press the "quote" key (`'`), it puts quotes around the selected word.
What would be great is to be able to unquote by pressing the same key if the selection already is already quoted.
It should work for simple quotes ('), double quotes (") and special quotes (`).
Thanks :)

**Comment by ThatXliner** [comment]
Add the following to your init script:

```coffeescript
atom.commands.add 'atom-text-editor', 'custom:unquote', (event) ->
    editor = atom.workspace.getActiveTextEditor()
    if editor != undefined
        editor.mutateSelectedText (sel) ->
            text = sel.getText()
            if text[0] == text.slice(-1) and '"\'`'.includes(text[0])
                sel.insertText(text.slice(1, -1))
```

 - If you want to **run** this custom command, go to the [command palette](https://flight-manual.atom.io/getting-started/sections/atom-basics/#command-palette) and search for `custom:unquote`
 - If you want to **bind** this command to a shortcut, see [Customizing Keybindings](https://flight-manual.atom.io/using-atom/sections/basic-customization/#customizing-keybindings)

**Reply by ThatXliner** [reply]
It should work now

**Reply by djcaesar9114** [reply]
Thanks, it works but I'd like to unquote automatically, atom detecting which quotes I want to remove.

**Reply by ThatXliner** [reply]
This should unquote stuff being quoted with the following characters:

 - `"`
 - `'`
 - <code>`</code>

I'm not sure what you mean by "atom detecting which quotes I want to remove"

**Reply by djcaesar9114** [reply]
For example if I have `"hello"` and I select the `hello` then either press:

- `"`: it should remove the quotes;
- `'`: it should add `'` quotes around `hello`, so that to make `"'hello'"`

If I select the whole `"hello"` then either press:

- `"`: it should add the quotes, so that to make `""hello""`;
- `'`: it should add `'` quotes around `"hello"`, so that to make `'"hello"'`
